backend/__init__.py

```diff
@@ -89,6 +108,39 @@ def create_app(env="backend.config.EnvironmentConfig"):
 
     app.logger.debug("Add root redirect route")
 
+    @app.errorhandler(Exception)
+    def handle_generic_error(error):
+        """Generic error handler for all exceptions"""
+        from backend.exceptions import format_sub_code
+
+        app.logger.exception(error)
+
+        error_message = (
+            str(error)
+            if len(str(error)) > 0
+            else ERROR_MESSAGES["INTERNAL_SERVER_ERROR"]
+        )
+        error_code = error.code if hasattr(error, "code") else 500
+        error_sub_code = (
+            format_sub_code(error.name)
+            if hasattr(error, "name")
+            else "INTERNAL_SERVER_ERROR"
+        )
+        return (
+            {
+                "error": {
+                    "code": error_code,
+                    "sub_code": error_sub_code,
+                    "message": error_message,
+                    "details": {
+                        "url": request.url,
+                        "method": request.method,
+                    },
+                }
+            },
+            error_code,
```

Comment:
Why the dict and the `error_code` also separately?

Reply:
Returned dict is response data while error_code is the HTTP status code associated with the error.